### DISCUSSION TITLE
Create role with id

### DISCUSSION THREAD

**Discussion by popdaniel942** [keycloak/keycloak#21984] [Q&A]
Hello. I've been trying to create a realm role through REST API while passing the "id" parameter with a uuid. But everytime the role gets created with another uuid. If I try creating a client while passing the "id", it actually uses that uuid for the client. But for roles it doesn't seem to work. Is this intended? I am using keycloak 20.0.2.
Thank you.

**Comment by sschu** [comment]
I would say this is intended. Why do you want to specify the ID for roles? Typically, this is only used internally...

**Reply by popdaniel942** [reply]
Well I want to implement fine grained permissions for some roles. From UI I can do it just fine. But I wanted to do it from a realm JSON which gets imported when keycloak is deployed. And for that I have to map in the JSON the roles with a policy. And this mapping is done though the role ID. But at deploy time, if I don't specify the role ID, I can't know what id it will have so I cannot map it.

**Reply by sschu** [reply]
OK, I got your use case. However, I fear currently this just won't work with a realm JSON. Only way I can imagine is by automating the creation of permissions using the API.
